tests/bsim/bluetooth/mesh/src/test_suspend.c

```diff
@@ -290,6 +408,131 @@ static void test_rx_suspend_disable_resume(void)
+	bt_mesh_adv_send(adv, NULL, NULL);
+	k_sleep(K_MSEC(150));
+
+	ASSERT_OK(bt_mesh_suspend());
+	LOG_INF("Mesh suspended.");
+
+	k_sleep(K_SECONDS(SUSPEND_DURATION));
+
+	ASSERT_OK(bt_mesh_resume());
+	LOG_INF("Mesh resumed.");
+
+	/* Send a mesh message to notify tester that device is resumed */
+	adv = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_ADV_TAG_LOCAL,
+			BT_MESH_TRANSMIT(0, 20), K_NO_WAIT);
+	net_buf_simple_add_mem(&adv->b, "0", sizeof("0"));
+	bt_mesh_adv_send(adv, NULL, NULL);
+	k_sleep(K_MSEC(150));
+
+	/* Let the tester observe that proxy advertisement resumes */
+	k_sleep(K_MSEC(6000));
+
+	PASS();
+}
+
+static void test_tx_gatt_suspend_disable_resume(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &comp);
+	ASSERT_OK_MSG(bt_mesh_prov_enable(BT_MESH_PROV_GATT), "Failed to enable GATT provisioner");
+	/* Let the tester observe pb gatt advertisements before provisioning. The node should
+	 * advertise pb gatt service with 100 msec (ext adv) or 1000msec (legacy adv) interval.
+	 */
+	k_sleep(K_MSEC(1800));
+
+	LOG_INF("Provisioning device under test");
+	bt_mesh_provision(test_net_key, 0, 0, 0, tx_cfg.addr, tx_cfg.dev_key);
+
+	/* Let the tester observe proxy advertisements */
+	k_sleep(K_MSEC(6500));
+
+	/* Send a mesh message to notify tester that device is suspended */
+	struct bt_mesh_adv *adv = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_ADV_TAG_LOCAL,
+			BT_MESH_TRANSMIT(0, 20), K_NO_WAIT);
+	net_buf_simple_add_mem(&adv->b, "1", sizeof("1"));
+	bt_mesh_adv_send(adv, NULL, NULL);
+	k_sleep(K_MSEC(150));
+
+	ASSERT_OK(bt_mesh_suspend());
+	LOG_INF("Mesh suspended.");
+
+	ASSERT_OK(bt_disable());
+	LOG_INF("Bluetooth disabled.");
+
+	k_sleep(K_SECONDS(SUSPEND_DURATION));
+
+	ASSERT_OK(bt_enable(NULL));
+	LOG_INF("Bluetooth enabled.");
+
+	ASSERT_OK(bt_mesh_resume());
+	LOG_INF("Mesh resumed.");
+
+	/* Send a mesh message to notify tester that device is resumed */
+	adv = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_ADV_TAG_LOCAL,
+			BT_MESH_TRANSMIT(0, 20), K_NO_WAIT);
+	net_buf_simple_add_mem(&adv->b, "0", sizeof("0"));
+	bt_mesh_adv_send(adv, NULL, NULL);
+	k_sleep(K_MSEC(150));
+
+	/* Let the tester observe that proxy advertisement resumes */
+	k_sleep(K_MSEC(6000));
+
+	PASS();
+}
+
+static void test_rx_gatt(void)
+{
+	k_sem_init(&publish_sem, 0, 1);
+
+	ASSERT_OK_MSG(bt_enable(NULL), "Bluetooth init failed");
+	LOG_INF("Bluetooth initialized");
+
+	/* Scan pb gatt beacons. */
+	gatt_rx_beacons();
+
+	/* Delay to provision dut */
+	k_sleep(K_MSEC(1000));
+
+	/* Scan gatt proxy beacons. */
+	gatt_param.transmits = 5000 / 1000;
+	gatt_param.interval = 1000;
+	gatt_param.service = MESH_SERVICE_PROXY;
+	gatt_rx_beacons();
+
+	/* Allow dut to suspend before scanning for gatt proxy beacons */
+	gatt_rx_xmit_adv();
+	k_sleep(K_MSEC(500));
+	gatt_rx_beacons();
+
+	/* Wait for dut to resume Mesh and notify tester, then scan for gatt proxy beacons */
```

Comment:
```suggestion
	/* Wait for DUT to resume Mesh and notify Tester, then scan for gatt proxy beacons */
```

Since we are talking of DUT and tester, I would prefer if you updated the test declaration. I know `tx` and `rx` are widely used, but in this instance I belive that `dut` and `tester` provides better insight for the reader. Also, I'd prefer something a bit more descriptive in the test string. Maybe something like this?:
```
static const struct bst_test_instance test_suspend[] = {
	TEST_CASE(dut, suspend_resume, "Suspend and resume Mesh while advertising periodic pub"),
	TEST_CASE(dut, suspend_disable_resume,
		  "Suspend and resume Mesh (and disable/enable BT) while advertising periodic pub"),
	TEST_CASE(dut, gatt_suspend_resume, "Suspend and resume Mesh while advertising GATT proxy"),
	TEST_CASE(dut, gatt_suspend_disable_resume,
		  "Suspend and resume Mesh (and disable/enable BT) while advertising GATT proxy"),

	TEST_CASE(tester, pub, "Scan and verify behavior of periodic pub adv"),
	TEST_CASE(tester, gatt, "Scan and verify behavior of GATT proxy adv"),

	BSTEST_END_MARKER};
```